source/extensions/filters/http/ext_proc/client.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <chrono>
+#include <memory>
+
+#include "envoy/grpc/status.h"
+#include "envoy/service/ext_proc/v3alpha/external_processor.pb.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ExternalProcessing {
+
+class ExternalProcessorStream {
+public:
+  virtual ~ExternalProcessorStream() = default;
+  virtual void send(envoy::service::ext_proc::v3alpha::ProcessingRequest&& request,
+                    bool end_stream) = 0;
+  virtual void close() = 0;
+};
+
+using ExternalProcessorStreamPtr = std::unique_ptr<ExternalProcessorStream>;
+
+class ExternalProcessorCallbacks {
```

Comment:
Does this wrapper layer give you much over the existing stream callbacks? Maybe you are planning on doing more in here later?

Reply:
Good point. In general I wanted to have a clean abstraction to ease unit testing. One difference is that I don't think that we'll be needing to handle the metadata callbacks and this abstraction takes care of them for me.